src/averageAges.js

```diff
@@ -55,10 +73,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleWithMothers = people.filter((person) => {
+    return (onlyWithSon
+      ? (person.sex === MALE)
+      : true
+    ) && people.some((mom) => mom.name === person.mother);
+  });
+
+  return peopleWithMothers
+    .reduce((acc, person) => acc + person.born - people
+      .find((mom) => mom.name === person.mother).born, 0)
+    / peopleWithMothers.length;
```

Comment:
Don't write this into one line, it's difficult to read. Better will be to create variable for `.find((mom) => mom.name === person.mother).born`